libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -0,0 +1,139 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * A raw result of parsing media types from Accept or Content-Type headers.
+ * It follow parsing and validates as per  rules defined in https://tools.ietf.org/html/rfc7231#section-3.1.1.1
+ * Can be resolved to <code>MediaType</code>
+ * @see MediaType
+ * @see MediaTypeRegistry
+ */
+public class ParsedMediaType {
+    // TODO this should be removed once strict parsing is implemented https://github.com/elastic/elasticsearch/issues/63080
+    // sun.net.www.protocol.http.HttpURLConnection sets a default Accept header if it was not provided on a request.
+    // For this value Parsing returns null.
+    public static final String DEFAULT_ACCEPT_STRING = "text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2";
+
+    private final String type;
+    private final String subType;
+    private final Map<String, String> parameters;
+    // tchar pattern as defined by RFC7230 section 3.2.6
+    private static final Pattern TCHAR_PATTERN = Pattern.compile("[a-zA-z0-9!#$%&'*+\\-.\\^_`|~]+");
+
+    private ParsedMediaType(String type, String subType, Map<String, String> parameters) {
+        this.type = type;
+        this.subType = subType;
+        this.parameters = Collections.unmodifiableMap(parameters);
+    }
+
+    /**
+     * The parsed mime type without the associated parameters. Will always return lowercase.
+     */
+    public String mediaTypeWithoutParameters() {
+        return type + "/" + subType;
+    }
+
+    public Map<String, String> getParameters() {
+        return parameters;
+    }
+
+    /**
+     * Parses a header value into it's parts.
+     * Note: parsing can return null, but it will throw exceptions once https://github.com/elastic/elasticsearch/issues/63080 is done
+     * Do not rely on nulls
+     *
+     * @return a {@link ParsedMediaType} if the header could be parsed.
+     * @throws IllegalArgumentException if the header is malformed
+     */
+    public static ParsedMediaType parseMediaType(String headerValue) {
+        if (DEFAULT_ACCEPT_STRING.equals(headerValue) || "*/*".equals(headerValue)) {
```

Comment:
should `*/*` be a JSON type ? (with this lenient parsing null defaults to that..but I think it should be explicit that `*/*` is JSON)

Reply:
It is a JSON type but only when used on Accept header. 
`*/*` or `text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2` are used when a user did not set an Accept header, so I think it is good that it results in null from `parseMediaType`. The same would be returned if header indeed was not set via some other tooling.

We can think of creating a new Media type where */* or `default java accept header` maps to JSON. We then add it to default `validAcceptMediaTypes`.

For this PR I think we stay with the lenience that exist now.
I added a note to https://github.com/elastic/elasticsearch/issues/63080